core/txindexer.go

```diff
@@ -99,33 +96,27 @@ func (indexer *txIndexer) run(tail *uint64, head uint64, stop chan struct{}, don
 	}
 
 	// Defensively ensure tail is not nil.
-	if tail == nil {
-		tail = new(uint64)
+	tailValue := uint64(0)
```

Comment:
Why the change here? It's effectively the same thing.

Reply:
it avoids overriding pointer